### ISSUE TITLE
Virtual properties getter not proxied

### ISSUE THREAD

**Issue by wolfeno** [mikeedwards83/Glass.Mapper#370]
In version 5.3 of Glass.Mapper.Sc having following models code
```
        [SitecoreType(TemplateId = TemplateId, AutoMap = true)]
        public class SampleClass 
        {
            [SitecoreInfo(SitecoreInfoType.Name)]
            public virtual string Name { get; set; }

            [SitecoreInfo(SitecoreInfoType.DisplayName)]
            public virtual string DisplayName { get; set; }

            public virtual string AnotherName => Name + DisplayName;
        }
```

After querying collection of items from Sitecore mapped to SampleClass AnotherName property is null on Castle proxy object even if both Name and DisplayName have value.

Reproduction case:
```
public class SitecoreServiceFixture
    {
        [Test]
        public void GetItem_IgnoredPropertyWithLogic_ShouldHaveValue()
        {
            //Assign


            Guid id = Guid.NewGuid();

            using (Db database = new Db
            {
                new Sitecore.FakeDb.DbItem("Target", new ID(id))
                {
                    Name = "AAAAA"
                }
            })
            {
                var context = Context.Create(Utilities.CreateStandardResolver());
                context.Load(new OnDemandLoader<SitecoreTypeConfiguration>(typeof(StubClass)));


                var service = new SitecoreService(database.Database);

                //Act
                var result = service.GetItem<StubClass>(id);

                //Assert
                Assert.IsNotNull(result);
                Assert.AreEqual("AAAAA1", result.Somefield);
                Assert.AreEqual("AAAAA1", result.Anotherfield);
            }
        }

    }

    [SitecoreType(TemplateId = TemplateId, AutoMap = true)]
    public class StubClass
    {
        public const string TemplateId = "{ABE81623-6250-46F3-914C-6926697B9A86}";

        [SitecoreId]
        public virtual Guid Id { get; set; }

        [SitecoreInfo(SitecoreInfoType.Version)]
        public virtual int Version { get; set; }

        [SitecoreInfo(SitecoreInfoType.Name)]
        public virtual string Name { get; set; }

        [SitecoreIgnore]
        public virtual string Somefield => Name + Version;

        public virtual string Anotherfield => Name + Version;
    }
```

**Comment by mikeedwards83** [comment]
I have pushed a fix into the Develop branch for this problem. I will be in the next release

**Comment by mikeedwards83** [comment]
Fixed in release 5.3.17
